### PR TITLE
[ntsc-1.2] Match __osMalloc.c and code_800FC620.c (new/delete)

### PR DESCRIPTION
**Pull request by cadmic**
__osMalloc got a bit unwieldy as one file so I've also split it into separate __osMalloc_n64.c and __osMalloc_gc.c files, like fault. I tried to keep the same macros though to make the similarities clear (for example, __osMalloc in N64 is identical to __osMalloc_NoLock in GC with the macros).

I removed FAULT_VERSION because the N64 fault implementation depends on gTotalAllocFailures, which doesn't exist in GC debug, so the versions of fault and __osMalloc seem linked. I think in the future it would be better to name the library or set of libraries these came from, and version that whole set together. This includes rand.c, so I've removed RAND_VERSION as well. To deal with compilation, I've wrapped the entire _n64.c/_gc.c files in #if PLATFORM_N64/PLATFORM_GC.

I took the opportunity to factor out the declarations into a new __osMalloc.h header. code_800FC620 does the same interrupt disabling as system_malloc.c, so I moved those macros into __osMalloc.h as well.

### CODE REVIEW

**Comment by cadmic** [comment]
Turns out I messed up the data splits for __osMalloc, so I've fixed those and added a bunch of unused strings. Not sure if we should make yet another printf macro for these. Also, the string `"$Id: sprintf.c,v 1.5 1997/03/19 02:28:53 hayakawa Exp $"` has appeared

File: include/__osMalloc.h
Comment (fig02) [inline include/__osMalloc.h:1]: I think osMalloc is good for the header name, I dont think it needs the __ that the functions have

**Review by fig02 — APPROVED** [review]

**Review by krm01 — APPROVED** [review]
its not part of this PR so still approving, just the first time i'm seeing it, but this `DECLARE_INTERRUPT_MASK` thing is pretty ugly, just floating there with no semicolon. really no way around that huh? i guess at least its localized to just that file.

**Comment by cadmic** [comment]
Yeah I don't of anything that can go before a semicolon here that wouldn't create unwanted stack padding. There's always the option of getting rid of these macros and ifdef-ing every function that uses them (in system_malloc.c and code_800FC620.c)

**Pull request merged**